src/llm/visual_language_model/legacy/servable.cpp

```diff
@@ -0,0 +1,195 @@
+    }
+    return absl::OkStatus();
+}
+
+absl::Status VisualLanguageModelLegacyServable::scheduleExecution(std::shared_ptr<GenAiServableExecutionContext>& executionContext) {
+    auto legacyExecutionContext = std::static_pointer_cast<VisualLanguageModelLegacyServableExecutionContext>(executionContext);
+    if (legacyExecutionContext->payload.client->isDisconnected()) {
+        return absl::CancelledError();
+    }
+    properties->legacyExecutor->addRequest(legacyExecutionContext);
+    return absl::OkStatus();
+}
+
+absl::Status VisualLanguageModelLegacyServable::readCompleteExecutionResults(std::shared_ptr<GenAiServableExecutionContext>& executionContext) {
+    auto legacyExecutionContext = std::static_pointer_cast<VisualLanguageModelLegacyServableExecutionContext>(executionContext);
+    if (legacyExecutionContext->payload.client->isDisconnected()) {
+        return absl::CancelledError();
+    }
+    legacyExecutionContext->finished.wait();
+    return absl::OkStatus();
+}
+
+absl::Status VisualLanguageModelLegacyServable::prepareCompleteResponse(std::shared_ptr<GenAiServableExecutionContext>& executionContext) {
+    auto legacyExecutionContext = std::static_pointer_cast<VisualLanguageModelLegacyServableExecutionContext>(executionContext);
+    if (legacyExecutionContext->payload.client->isDisconnected()) {
+        return absl::CancelledError();
+    }
+    executionContext->response = executionContext->apiHandler->serializeUnaryResponse(legacyExecutionContext->results);
+    SPDLOG_LOGGER_DEBUG(llm_calculator_logger, "Complete unary response: {}", executionContext->response);
+    return absl::OkStatus();
+}
+
+absl::Status VisualLanguageModelLegacyServable::readPartialExecutionResults(std::shared_ptr<GenAiServableExecutionContext>& executionContext) {
+    return absl::OkStatus();
+}
+
+absl::Status VisualLanguageModelLegacyServable::preparePartialResponse(std::shared_ptr<GenAiServableExecutionContext>& executionContext) {
+    auto legacyExecutionContext = std::static_pointer_cast<VisualLanguageModelLegacyServableExecutionContext>(executionContext);
+    if (legacyExecutionContext->payload.client->isDisconnected()) {
+        return absl::CancelledError();
+    }
+    std::string lastTextChunk;
+    auto generationStatus = legacyExecutionContext->finished.wait_for(std::chrono::nanoseconds::zero());
+    {
+        std::unique_lock lock(legacyExecutionContext->mutex);
+        while (executionContext->lastStreamerCallbackOutput.size() == 0 && generationStatus != std::future_status::ready) {
+            SPDLOG_LOGGER_TRACE(llm_executor_logger, "Waiting for partial data...");
+            legacyExecutionContext->executionInProgress.wait(lock);
+            generationStatus = legacyExecutionContext->finished.wait_for(std::chrono::nanoseconds::zero());
+        }
+        lastTextChunk = executionContext->lastStreamerCallbackOutput;
```

Comment:
We can properly handle usage parameter here if user requests it. You can add below:
```
        if (!lastTextChunk.empty()) {
            auto tokensTensor = properties->tokenizer.encode(lastTextChunk, ov::genai::add_special_tokens(false)).input_ids;
            auto numTokens = tokensTensor.get_size();
            executionContext->apiHandler->incrementProcessedTokens(numTokens);
```